### ISSUE TITLE
[Feature]: (Info) Confusing terminology related to sweep transactions 

### ISSUE THREAD

**Issue by technovision99** [stacks-sbtc/sbtc#986]
## 1. Description
The deposit flow currently uses `sweep` and `sbtc_transactions` interchangeably which is suboptimal for a clear understanding. For example, `extract_sbtc_transactions` in the block_observer monitors for potential sweep transactions and the terminology should be aligned clearer.
Sweep transactions are committed to the database before actual verification. The terminology here should also be clearer since these are actually unconfirmed sweep transactions at this stage and only actually verified later on by the transaction signer.

**Comment by cylewitruk** [comment]
I agree -- the term "sweep" came about after-the-fact and we're behind in synchronizing the terminology, so good call-out.

`extract_sbtc_transactions` however does also look for non-sweep transactions, such as transactions including both donations and deposits, which are not sweep transactions, so they are not necessarily interchangeable. I generally read an `sBTC transaction` to be any transaction which contains sBTC-related information in it, whereas a `sweep` is specifically referring to signer-created transactions (either sweeping-in deposits our sweeping-out withdrawals).

**Comment by evonide** [comment]
Thanks Cyle,
> extract_sbtc_transactions however does also look for non-sweep transactions, such as transactions including both donations and deposits, which are not sweep transactions

Currently, [`extract_sbtc_transactions`](https://github.com/stacks-network/sbtc/blob/e647881c744be3e2034da7c69b45c33123ac516b/signer/src/block_observer.rs#L365) seems to only look for BTC outputs paid directly to signer script pubkeys. It doesn't detect the user deposit transactions that are spendable via the deposit script path. As you mentioned, apart from actual sweep transactions they can of course also be donations as pointed out. A more intuitive name could be something like `extract_signer_spendable_transactions`. The actual BTC deposits get processed via `load_latest_deposit_requests()` by [pulling the open requests from the Emily API](https://github.com/stacks-network/sbtc/blob/e647881c744be3e2034da7c69b45c33123ac516b/signer/src/block_observer.rs#L180) and then [invoking the Bitcoin client for validation](https://github.com/stacks-network/sbtc/blob/e647881c744be3e2034da7c69b45c33123ac516b/signer/src/block_observer.rs#L70).

However, the mai thing to consider refactoring is when the system actually commits sweep deposits to the database. Currently, the transaction coordinator [issues the BTC sweep transaction](https://github.com/stacks-network/sbtc/blob/e647881c744be3e2034da7c69b45c33123ac516b/signer/src/transaction_coordinator.rs#L848-L851) and then [communicates that this has happened to the network peers via `SweepTransactionInfo`](https://github.com/stacks-network/sbtc/blob/e647881c744be3e2034da7c69b45c33123ac516b/signer/src/transaction_coordinator.rs#L850). The transaction signer's `handle_signer_message()` code then processes such a `message::Payload::SweepTransactionInfo` and writes it immediately to the database via https://github.com/stacks-network/sbtc/blob/e647881c744be3e2034da7c69b45c33123ac516b/signer/src/transaction_signer.rs#L271-L274

Note that the sweep transaction is unconfirmed at this point as it can reside exclusively in the mempool. The actual verification of this sweep transaction does happen but only a later stage making this part more difficult to understand. The improvement suggestion to consider would be to explicitly call them out as "pending" at this stage.

**Comment by cylewitruk** [comment]
Yeah, sorry, you're right -- we've been discussing this on the side. Some of this is currently in a state of flux, and it's possible that the `sweep_transactions` table along with `SweepTransactionInfo` message will be removed in favor of storing only confirmed information directly from the blockchain (which we're doing now in `extract_sbtc_transactions()` by storing the related inputs into `bitcoin_tx_inputs` for sweep transactions, including deposit prevouts).

The `SweepTransactionInfo` was largely intended for fee calculation in the case of RBF of sweep transactions, but we've changed the coordinator now to read these transactions directly from the mempool and calculate fees based on that instead of relying on unconfirmed state reported over gossipsub which may or may not match the local signer's view of the mempool.

**Comment by djordon** [comment]
> The deposit flow currently uses `sweep` and `sbtc_transactions` interchangeably which is suboptimal for a clear understanding. For example, `extract_sbtc_transactions` in the block_observer monitors for potential sweep transactions and the terminology should be aligned clearer.

Yeah, true. Some terminology should be updated in the repo.



> Sweep transactions are committed to the database before actual verification. The terminology here should also be clearer since these are actually unconfirmed sweep transactions at this stage and only actually verified later on by the transaction signer.

We no longer commit unverified information to the database. Also, the logic for writing unverified sweep transactions to the database was removed before launch in https://github.com/stacks-sbtc/sbtc/pull/1060.

Closing this, since we are not going to take action on this anytime soon.
